applications/GeoMechanicsApplication/custom_elements/line_2d_5.hpp

```diff
@@ -0,0 +1,1314 @@
+         * Calculates the gradients in terms of local coordinates
+         * of all shape functions in a given point.
+         *
+         * @param rPoint the current point at which the gradients are calculated
+         * @return the gradients of all shape functions
+         * \f$ \frac{\partial N^i}{\partial \xi_j} \f$
+         */
+        Matrix& ShapeFunctionsLocalGradients(Matrix& rResult, const CoordinatesArrayType& rPoint) const override
+        {
+            // Setting up result matrix
+            if (rResult.size1() != 5 || rResult.size2() != 1) rResult.resize(5, 1, false);
+            noalias(rResult) = ZeroMatrix(5, 1);
+            //
+            double xcor = rPoint[0];
+            double fx1 = xcor - 1.0;
+            double fx2 = xcor + 1.0;
+            double fx3 = fx1 * fx2;
+            double gx1 = 2.0 * xcor - 1.0;
+            double gx2 = 2.0 * xcor + 1.0;
+            double gx3 = gx1 * gx2;
+            double cof1 = 1.0 / 6.0;
+            double cof2 = 8.0 * cof1;
+            //
+            rResult(0, 0) = gx3 * gx1 * cof1 + xcor * xcor * fx1 * cof2;
+            rResult(1, 0) = gx3 * gx2 * cof1 + xcor * xcor * fx2 * cof2;
+            rResult(2, 0) = -(gx1 * (gx3 - fx3 - 1.0) + 2.0 * xcor * fx3) * cof2;
+            rResult(3, 0) = 2.0 * xcor * (gx3 + 4.0 * fx3);
+            rResult(4, 0) = -(gx2 * (gx3 - fx3 - 1.0) + 2.0 * xcor * fx3) * cof2;
+            return(rResult);
+        }
+
+        // ========================================================================================
+        // ========================================================================================
+        /**
+         * returns the local coordinates of all nodes of the current geometry
+         * @param rResult a Matrix object that will be overwritten by the result
+         * @return the local coordinates of all nodes
+         */
+        Matrix& PointsLocalCoordinates(Matrix& rResult) const override
+        {
+            if (rResult.size1() != 5 || rResult.size2() != 1) rResult.resize(5, 1, false);
+            noalias(rResult) = ZeroMatrix(5, 1);
+            rResult(0, 0) = -1.0;
+            rResult(1, 0) = 1.0;
+            rResult(2, 0) = -0.5;
+            rResult(3, 0) = 0.0;
+            rResult(4, 0) = 0.5;
+            return rResult;
+        }
+
+        // ========================================================================================
+        // ========================================================================================
+        /**
+         * returns the shape function gradients in an arbitrary point,
+         * given in local coordinates
+         * @param rResult the matrix of gradients, will be overwritten
+         * with the gradients for all
+         * shape functions in given point
+         * @param rPoint the given point the gradients are calculated in
+         */
+        virtual Matrix& ShapeFunctionsGradients(Matrix& rResult, CoordinatesArrayType& rPoint)
```

Comment:
const CoordinatesArrayType& rPoint?

Reply:
It seems that rPoint is constant in this function.

Reply:
Done